command/agent/config.go

```diff
@@ -1072,6 +1086,24 @@ func (a *Telemetry) PrefixFilters() (allowed, blocked []string, err error) {
 	return allowed, blocked, nil
 }
 
+// Validate the telemetry configuration options. These are used by the agent,
+// regardless of mode, so can live here rather than a structs package. It is
+// safe to call, without checking whether the config object is nil first.
+func (t *Telemetry) Validate() error {
+	if t == nil {
+		return nil
+	}
+
+	// Ensure the in-memory durations are set correctly.
+	if t.inMemoryRetentionPeriod > 0 && t.inMemoryCollectionInterval > 0 {
+		if t.inMemoryCollectionInterval > t.inMemoryRetentionPeriod {
+			return errors.New("telemetry in-memory collection interval cannot be greater than retention period")
+		}
+	}
+
+	return nil
+}
```

Comment:
Should we check for 0 values here? From my quick test it seems that a `0` uses the default value. Not sure if that's expected though.

Reply:
Added within: https://github.com/hashicorp/nomad/pull/20166/commits/308f0a1fa64b1331d8b3d5de0e1622196b97abc1